engine/src/test/java/org/camunda/bpm/engine/test/api/mgmt/license/LicenseKeyTest.java

```diff
@@ -0,0 +1,164 @@
+
+public class LicenseKeyTest {
+
+  public ProvidedProcessEngineRule engineRule = new ProvidedProcessEngineRule();
+  public ProcessEngineTestRule testRule = new ProcessEngineTestRule(engineRule);
+
+  @Rule
+  public RuleChain ruleChain = RuleChain.outerRule(testRule).around(engineRule);
+
+  ProcessEngine processEngine;
+  ProcessEngineConfigurationImpl processEngineConfiguration;
+  ManagementService managementService;
+
+  @Before
+  public void init() {
+    processEngine = engineRule.getProcessEngine();
+    processEngineConfiguration = (ProcessEngineConfigurationImpl) processEngine.getProcessEngineConfiguration();
+    managementService = processEngine.getManagementService();
+  }
+
+  @After
+  public void tearDown() {
+    managementService.deleteLicenseKey();
+  }
+
+  @Test
+  public void testSetLicenseKey() {
+    // given
+    String licenseKey = "testLicenseKey";
+
+    // when
+    managementService.setLicenseKey(licenseKey);
+    Map<String, String> properties = managementService.getProperties();
+    String licenseKeyLegacyProperty = properties.get(LicenseCmd.LICENSE_KEY_PROPERTY_NAME);
+    String licenseByteArrayId = properties.get(LicenseCmd.LICENSE_KEY_BYTE_ARRAY_ID);
+
+    // then
+    assertThat(licenseKeyLegacyProperty).isNull();
+    assertThat(licenseByteArrayId).isNotNull();
+    // make sure a newly set license is not stored in properties...
+    assertThat(licenseByteArrayId).isNotEqualTo(licenseKey);
+    // ...but in the byte array table, referenced by the property
+    assertThat(processEngineConfiguration.getCommandExecutorTxRequired().execute(new GetByteArrayCommand(licenseByteArrayId)).getBytes())
+        .isEqualTo(licenseKey.getBytes());
+  }
+
+  @Test
+  public void testSetDuplicateLicenseKey() {
+    // given
+    String licenseKey = "testLicenseKey";
+    String licenseKey2 = "testLicenseKey2";
+
+    // when
+    managementService.setLicenseKey(licenseKey);
+    String storedLicenseKey = managementService.getLicenseKey();
+    managementService.setLicenseKey(licenseKey2);
+    String storedLicenseKey2 = managementService.getLicenseKey();
+
+    // then
+    assertThat(storedLicenseKey).isEqualTo(licenseKey);
+    assertThat(storedLicenseKey2).isEqualTo(licenseKey2);
+    assertThat(storedLicenseKey).isNotEqualTo(storedLicenseKey2);
+  }
+
+  @Test
+  public void testGetLicenseKey() {
+    // given
+    String legacyLicenseKey = "testLegacyLicenseKey";
+    String licenseKey = "testLicenseKey";
+    managementService.setProperty(LicenseCmd.LICENSE_KEY_PROPERTY_NAME, legacyLicenseKey);
+
+    // when
+    String storedLegacyLicenseKey = managementService.getLicenseKey();
+    managementService.setLicenseKey(licenseKey);
+    String storedLicenseKey = managementService.getLicenseKey();
+
+    // then
+    assertThat(storedLegacyLicenseKey).isEqualTo(legacyLicenseKey);
+    assertThat(storedLicenseKey).isEqualTo(licenseKey);
+  }
+
+  @Test
+  public void testDeleteLicenseKey() {
```

Comment:
1. Please keep the test as focused as possible and split it up, meaning that we should test 
   1. "given we set a license key via `managementService#setLicenseKey`, we can delete it via `managementService#deleteLicenseKey`
   1. "given we set a legacy license key via `managementService#setProperty`, we can delete it via `managementService#deleteLicenseKey` as well
1. We should also test that we can call the delete when no key is set in the engine

Reply:
I agree